core/chains/evm/client/node_lifecycle.go

```diff
@@ -173,6 +173,9 @@ func (n *node) aliveLoop() {
 				lggr.Tracew("Ignoring previously seen block number", "latestReceivedBlockNumber", latestReceivedBlockNumber, "blockNumber", bh.Number, "nodeState", n.State())
 			}
 			outOfSyncT.Reset(noNewHeadsTimeoutThreshold)
+			n.stateMu.Lock()
+			n.latestReceivedBlockNumber = bh.Number
```

Comment:
the received number can be lower, and we still update the latest, which will help us to treat this node as potentially out of sync